src/Client/BuzzHouse/Generator/ExternalIntegrations.cpp

```diff
@@ -16,124 +16,555 @@
 namespace BuzzHouse
 {
 
-bool MinIOIntegration::sendRequest(const std::string & resource)
+bool ClickHouseIntegratedDatabase::performIntegration(
+    RandomGenerator & rg,
+    std::shared_ptr<SQLDatabase> db,
+    const uint32_t tname,
+    const bool can_shuffle,
+    std::vector<ColumnPathChain> & entries)
 {
-    struct tm ttm;
-    ssize_t nbytes = 0;
-    bool created = false;
-    int sock = -1;
-    int error = 0;
-    char buffer[1024];
-    char found_ip[1024];
-    const std::time_t time = std::time({});
-    DB::WriteBufferFromOwnString sign_cmd;
-    DB::WriteBufferFromOwnString sign_out;
-    DB::WriteBufferFromOwnString sign_err;
-    DB::WriteBufferFromOwnString http_request;
-    struct addrinfo hints = {};
-    struct addrinfo * result = nullptr;
-
-    hints.ai_family = AF_INET;
-    hints.ai_socktype = SOCK_STREAM;
-    hints.ai_protocol = IPPROTO_TCP;
+    const String str_tname = getTableName(db, tname);
 
-    (void)std::sprintf(buffer, "%" PRIu32 "", sc.port);
-    if ((error = getaddrinfo(sc.hostname.c_str(), buffer, &hints, &result)) != 0)
+    if (performQuery(fmt::format("DROP TABLE IF EXISTS {};", str_tname)))
     {
-        if (error == EAI_SYSTEM)
+        String buf;
+        bool first = true;
+
+        buf += "CREATE TABLE ";
+        buf += str_tname;
+        buf += "(";
+
+        if (can_shuffle && rg.nextSmallNumber() < 7)
         {
-            strerror_r(errno, buffer, sizeof(buffer));
-            LOG_ERROR(fc.log, "getnameinfo error: {}", buffer);
+            std::shuffle(entries.begin(), entries.end(), rg.generator);
         }
-        else
+        for (const auto & entry : entries)
         {
-            LOG_ERROR(fc.log, "getnameinfo error: {}", gai_strerror(error));
+            SQLType * tp = entry.getBottomType();
+
+            if (!first)
+            {
+                buf += ", ";
+            }
+            buf += entry.getBottomName();
+            buf += " ";
+            buf += columnTypeAsString(rg, tp);
+            buf += " ";
+            buf += ((entry.nullable.has_value() && entry.nullable.value()) || hasType<Nullable>(false, false, false, tp)) ? "" : "NOT ";
+            buf += "NULL";
+            assert(entry.path.size() == 1);
+            first = false;
         }
-        return false;
+        buf += ");";
```

Comment:
I still find these kind of String concatenation using `+=` incredibly ugly because it makes harder to understand the final String generated. I still recommend using `fmt::format` or `std::format` instead

Reply:
This should be fixed now